.github/workflows/build-and-test.yml

```
@@ -37,7 +37,6 @@ jobs:
       - name: Install Rust
         uses: dtolnay/rust-toolchain@38b70195107dddab2c7bbd522bcf763bac00963b # stable
```

Let's pin the action to the same version as in `Cargo.toml` ([1.90](https://github.com/dtolnay/rust-toolchain/tree/1.90)), and hopefully Dependabot would open PRs to update this version:

```suggestion
        uses: dtolnay/rust-toolchain@9a1d20035bdbcbc899baabe1e402e85bc33639bc # v1.90
```

See: https://github.com/dtolnay/rust-toolchain/tree/1.90?tab=readme-ov-file#install-rust-toolchain

> The selection of Rust toolchain is made based on the particular `@rev` of this Action being requested. For example "dtolnay/rust-toolchain@nightly" pulls in the nightly Rust toolchain, while "dtolnay/rust-toolchain@1.89.0" pulls in 1.89.0.